### ISSUE TITLE
Move build_targets and process_target to tough

### ISSUE THREAD

**Issue by webern** [awslabs/tough#153]
I found when using tough to build a repository with the new RepositoryEditor (in the absence of tuftool) that when I really wanted these functions in tough.

```rust
// Walk the directory specified, building a map of filename to Target structs.
// Hashing of the targets is done in parallel
fn build_targets<P>(indir: P, follow_links: bool) -> Result<HashMap<String, Target>>
where
    P: AsRef<Path>,
{
    let indir = indir.as_ref();
    WalkDir::new(indir)
        .follow_links(follow_links)
        .into_iter()
        .par_bridge()
        .filter_map(|entry| match entry {
            Ok(entry) => {
                if entry.file_type().is_file() {
                    Some(process_target(entry.path()))
                } else {
                    None
                }
            }
            Err(err) => Some(Err(err).context(error::WalkDir { directory: indir })),
        })
        .collect()
}

fn process_target(path: &Path) -> Result<(String, Target)> {
    // Build a Target from the path given. If it is not a file, this will fail
    let target = Target::from_path(path).context(error::TargetFromPath { path })?;

    // Get the file name as a string
    let target_name = path
        .file_name()
        .context(error::NoFileName { path })?
        .to_str()
        .context(error::PathUtf8 { path })?
        .to_owned();
    Ok((target_name, target))
}
```

**Comment by iliana** [comment]
Maybe something like `RepositoryEditor::add_targets_from_dir` would be useful? Perhaps in parallel with rayon as an optional feature :)

**Comment by webern** [comment]
> Maybe something like `RepositoryEditor::add_targets_from_dir` would be useful? Perhaps in parallel with rayon as an optional feature :)

Yeah, something like that. It would be nice to do it without the dependency unless it's unwieldy.

**Comment by zmrow** [comment]
@webern do you mean the dependency on `rayon`?  We use rayon to parallelize the hashing of targets and while it's entirely possible to do this hashing synchronously, it's nice to have if you have a decent amount of targets to add.

**Comment by webern** [comment]
i was just reacting with aversion to features which I sort of dislike. But if it makes more sense to make it a feature I won't stand in the way.
